mssql/resource_database.go

```diff
@@ -42,30 +55,46 @@ func resourceDatabaseRead(ctx context.Context, data *schema.ResourceData, meta i
 
 	var diags diag.Diagnostics
 
+	id := data.Get("id").(string)
+
+	client := meta.(*sqlcmd.SqlCommand)
+
+	database, err := client.GetDatabase(id, "database_id")
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	data.Set("name", database.Name)
+	data.Set("collation", database.Collation_name)
+	data.Set("recovery_mode", database.RecoveryModel)
+	data.Set("owner_sid", database.Ownersid)
+	data.Set("is_read_only", database.IsReadOnly)
```

Comment:
you want to consider keeping the code DRY,
this makes it much more maintainable.
in this case, move the setting of state (except id) to a single method
and call this method from your GET, CREATE, UPDATE (to prevent drift)

Reply:
Will do this in a separate pr because it is also used outside database (will change this for all resources)